invenio_rdm_records/records/dumpers.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""DataCite-based data model for Invenio."""
+
+import calendar
+from datetime import date
+
+from edtf import parse_edtf
+from invenio_records.dumpers import ElasticsearchDumperExt
+
+
+class EDTFDumperExt(ElasticsearchDumperExt):
+    """Elasticsearch dumper extension for EDTF dates support.
+
+    It adds two fields: publication_date_start and publication_date_end,
+    which correspond to the lower strict and upper strict bounds. This
+    are required for sorting. In an ascending sort (most recent last),
+    sort by publication_date_start to get a natural sort order. In a
+    descending sort (most recent first), sort by publication_date_end.
+    """
+
+    # NOTE: Hardcoded to publication_date. If the need for other EDTF fields
+    # is found this can be made configurable (e.g. fields list attribute.)
+
+    def dump(self, record, data):
```

Comment:
Minor: What if publication date is undefined? and what if there's a parsing error?

Reply:
Implemented, along with tests:

- The field does not exists (KeyError)
- The field contains an un-parseable string (EDTFParseException)

In both cases, the errors are caught and "ignored".